src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Models/RazorSemanticTokensLegend.cs

```diff
@@ -62,10 +63,24 @@ internal class RazorSemanticTokensLegend
     public static int CSharpVariable => TokenTypesLegend["variable"];
 
     // C# types + Razor types
-    public static readonly string[] TokenTypes = RazorSemanticTokensAccessor.RoslynTokenTypes.Concat(
-        typeof(RazorSemanticTokensLegend).GetFields(BindingFlags.NonPublic | BindingFlags.Static).Where(
-            field => field.GetValue(null) is string).Select(
-            field => (string)field.GetValue(null))).ToArray();
+    public static readonly ImmutableArray<string> TokenTypes = GetTokenTypes();
+
+    private static ImmutableArray<string> GetTokenTypes()
```

Comment:
It was done for a couple of reasons:

* Performance:
  * Cuts the number of `FieldInfo.GetValue(...)` calls in half. Removing unnecessary reflection calls is generally a good thing for performance. :smile:
* Correctness:
  * Using an array in this case is _slightly_ dangerous because an array can be modified. Code could set any `TokenTypes` element to a different value. It might seem super unlikely but using an `ImmutableArray<string>` here guards against programmer error.